cmake/Modules/Findfmt.cmake

```diff
@@ -0,0 +1,64 @@
+add_library(fmt INTERFACE IMPORTED)
+
+set(fmt_external_project_name fmtlib_fmt)
+
+find_path(fmt_INCLUDE_DIR fmt/format.h
+  PATHS ${EP_PREFIX}/src/${fmt_external_project_name}/include)
+mark_as_advanced(fmt_INCLUDE_DIR)
+
+# read the fmt version stored in fmt/core.h
+unset(fmt_VERSION)
+set(fmt_version_file_path ${fmt_INCLUDE_DIR}/fmt/core.h)
+if (EXISTS ${fmt_version_file_path})
+  file(READ ${fmt_version_file_path} fmt_version_file)
+  if (fmt_version_file MATCHES "FMT_VERSION ([0-9]+)([0-9][0-9])([0-9][0-9])")
+    # Use math to skip leading zeros if any.
+    math(EXPR ver_major ${CMAKE_MATCH_1})
+    math(EXPR ver_minor ${CMAKE_MATCH_2})
+    math(EXPR ver_patch ${CMAKE_MATCH_3})
+    set(fmt_VERSION "${ver_major}.${ver_minor}.${ver_patch}")
+  endif ()
+endif ()
+
+if (NOT DEFINED fmt_VERSION OR fmt_VERSION VERSION_LESS fmt_FIND_VERSION)
+  message(STATUS "Package 'fmt' of version ${fmt_FIND_VERSION} not found. "
+          "Will download it from git repo.")
+
+  set(git_url https://github.com/fmtlib/fmt.git)
+  set(fmt_VERSION ${fmt_FIND_VERSION})
+  set(git_tag "refs/tags/${fmt_VERSION}")
+  set_package_properties(fmt
+      PROPERTIES
+      URL ${git_url}
+      DESCRIPTION "String formatting library"
+      )
+
+  externalproject_add(${fmt_external_project_name}
+      GIT_REPOSITORY  ${git_url}
+      GIT_TAG         ${git_tag}
+      CONFIGURE_COMMAND "" # remove configure step
+      BUILD_COMMAND "" # remove build step
+      INSTALL_COMMAND "" # remove install step
+      TEST_COMMAND "" # remove test step
+      UPDATE_COMMAND "" # remove update step
```

Comment:
Most of it empty by default, so might be removed

Reply:
In [the docs](https://cmake.org/cmake/help/latest/module/ExternalProject.html) I can see that the default is to use the external project's CMake configuration. Both added libs use CMake, and both are used as header-only, so we have to explicitly disable the steps we do not need.